### PR TITLE
Add poetry2nix application for scheduler

### PR DESCRIPTION
**Pull request by steshaw**
Here we find the necessary poetry2nix overrides necessary to build a Poetry application for the scheduler.

Also included a fix for https://github.com/tweag/chainsail/issues/455 so that I can test the scheduler in Minikube.

### CODE REVIEW

File: app/scheduler/pyproject.toml [hunk 1]
```diff
@@ -32,14 +32,14 @@ uWSGI = "^2.0.20"
 cloudstorage = {extras = ["google"], version = "^0.11.0"}
 kubernetes = "^18.20.0"
 boto3 = "^1.20.26"
-setuptools = "^60.0.0"
+setuptools = "*"
```
Comment (steshaw) [inline app/scheduler/pyproject.toml:35]: @simeoncarstens, I hope that this doesn't cause any trouble with respect to https://github.com/tweag/chainsail/pull/437. Having "^60.0.0" meant that poetry could not solve the constraints.

File: app/scheduler/pyproject.toml [hunk 2]
```diff
@@ -32,14 +32,14 @@ uWSGI = "^2.0.20"
...
 
 [tool.poetry.dev-dependencies]
 pytest = "^6.2.1"
 black = "22.3.0"
 flake8 = "^3.8.4"
 pylint = "^2.6.0"
-mypy = "^0.790"
+mypy = "^1.0.0"
```
Comment (steshaw) [inline app/scheduler/pyproject.toml:42]: Upgraded mypy here, as in the `controller`.

File: app/scheduler/pyproject.toml (hunk 1 above)
Comment (simeoncarstens) [reply app/scheduler/pyproject.toml:35]: If it works, I guess it's fine. The lock file shows that the `setuptools` dependency resolves to v67.4, so very recent. And as such it's pinned in the lock file, but having no version bounds at all in `pyproject.toml` looks weird somehow.

**Comment by steshaw** [comment]
Note that I've rebased on changes that occurred in https://github.com/tweag/chainsail/pull/443.

**Comment by simeoncarstens** [comment]
> No, I didn't know how and so just relied on getting a green light from CI. I'll try to get a full k8s environment going so that I can get an end-to-end test working. It's good to be cautious because, with dynamically-typed languages, dependency upgrades like this are much more likely to break things.

Nice, thanks! Yes, we have to be extra-cautious with the controller, because the Flask app and the Celery tasks unfortunately don't have tests at the moment. Furthermore, the whole interplay between Celery / Redis / Flask app and the uwsgi web server setup are not covered by tests either, and I noticed #437 only when the Docker container wouldn't start.

**Comment by steshaw** [comment]
@simeoncarstens, Now that I've included a fix for #455, I've been able to test that the scheduler starts in Minikube. This is ready to merge if you're still happy with it.

**Comment by simeoncarstens** [comment]
That's excellent news, @steshaw! I would feel more at ease, though, if we were sure that also, e.g. Celery tasks work. I don't see a reason why they shouldn't, though, but just to be safe, we could wait until we fixed #449 (or temporarily hack a way around it), which should be all that's required for testing the Nix-packaged apps within the full service deployment.

**Comment by steshaw** [comment]
I can confirm that I was able to test an end-to-end workflow(which I assume exercises Celery tasks). To start the frontend dev server for the E2E workflow, I temporarily downgraded the nixpkgs pin to `nixpkgs/21.11`. I was able to create, start, and finish a job using the "mixture.zip" in the UI. I'm still experimenting with fixes to #449.

If you're happy to merge this on that basis. Perhaps you could then look at https://github.com/tweag/chainsail/pull/447 which is stacked on this PR.

**Pull request merged**